map.json

```diff
@@ -0,0 +1,117 @@
+[
+    {
+        "version": 2,
+        "path": "nvram3.bin",
+        "entries": [
+            {
+                "name": "_version",
+                "offset": 53,
+                "size": 1
+            },
+            {
+                "name": "save3_bool",
+                "offset": 54,
+                "size": 1,
+                "bit_offset": 0,
+                "is_bool": true
+            },
+            {
+                "name": "save3_bool2",
+                "offset": 54,
+                "size": 1,
+                "bit_offset": 1,
+                "is_bool": true
+            },
+            {
+                "name": "save3_byte",
+                "offset": 55,
+                "size": 1
+            },
+            {
+                "name": "save3_int",
+                "offset": 56,
+                "size": 4
+            },
+            {
+                "name": "save3_double",
+                "offset": 60,
+                "size": 8
+            },
+            {
+                "name": "save3_string",
+                "offset": 68,
+                "size": 2
+            },
+            {
+                "name": "save3_irange",
+                "offset": 70,
+                "size": 16
+            },
+            {
+                "name": "save3_drange",
+                "offset": 86,
+                "size": 32
```

Comment:
We can't check if they're OK while loading. If two variables overlap, how can they be checked? I'd just do something similar to what C does to structs, but specifying only the sizes; offsets would be calculated by the tool. Considering that this will be edited by hand and values will be added and removed, this is very likely to happen.